Library/Homebrew/brew.sh

```diff
@@ -247,7 +247,11 @@ then
   # Hide shellcheck complaint:
   # shellcheck source=/dev/null
   source "$HOMEBREW_BASH_COMMAND"
-  { update-preinstall; "homebrew-$HOMEBREW_COMMAND" "$@"; exit $?; }
```

Comment:
No, actually thanks for asking!

1. That's definitely behavior few shell users know about and in many situations they don't need to know about it. But in our case it's important and it's better to be safe.
2. Your comment made me think again about some other code in this PR and I think I identified a problem that has the potential to affect users. I'm investigating this now.